### PR TITLE
Identities offers both key and certificate when both are present

### LINKED ISSUE #452: Support for certificates and TrustedUserCAKeys
I have a scenario where I want my private key to be managed in a secure enclave by secretive, and I want to be able to include a signed certificate for my public key in my agent forwarding.

For example, when I configure using standard OpenSSH ssh-agent, I see this in `ssh-add -l`:

    ssh-add -l
    256 SHA256:xyz user@example.com (ECDSA)
    256 SHA256:xyz user@example.com (ECDSA-CERT)

But when I use Secretive, I don't see any way to include a certificate. So I only see:

    ssh-add -l
    256 SHA256:xyz user@example.com (ECDSA)

In practice this means that agent forwarding cannot work with hosts configured to use TrustedUserCAKeys.

Can secretive be somehow configured to support including certificates in available identities?

### PR DESCRIPTION
**Pull request by chriseldredge**
This PR modifies `Agent.identities()` to return certificates _in addition to_ keys, rather than returning only the certificate. This makes Secretive behave similarly to OpenSSH ssh-agent and addresses difficulties using the agent in scenarios where a private key can be used for authentication against a given remote server directly without the certificate.

Fixes #452.

### CODE REVIEW

**Comment by maxgoedjen** [comment]
@chriseldredge nice – thanks!

I'm not personally a certs user so I don't have direct experience in the mechanics of this one – is it correct to returns they key identities before the certs ones? My intuition would be that the certs should be offered first, but I'm really not sure there.

**Comment by chriseldredge** [comment]
I’m not an expert either but can observe what happens in my configuration. As noted in #452, OpenSSH lists the public key first, so I mimicked that order. 

I suspect this doesn’t matter much in practice, as most servers will either accept the certificate or the public key, usually not both, and in either situation the same private key gets used anyway.

**Comment by maxgoedjen** [comment]
> I’m not an expert either but can observe what happens in my configuration. As noted in https://github.com/maxgoedjen/secretive/issues/452, OpenSSH lists the public key first, so I mimicked that order.

Good enough for me, thanks!

File: Sources/Packages/Sources/SecretAgentKit/Agent.swift [hunk 1]
```diff
@@ -86,27 +86,24 @@ extension Agent {
...
-            if let (certificateData, name) = try? certificateHandler.keyBlobAndName(for: secret) {
-                keyBlob = certificateData
-                curveData = name
-            } else {
-                keyBlob = writer.data(secret: secret)
-                curveData = writer.curveType(for: secret.algorithm, length: secret.keySize).data(using: .utf8)!
-            }
-            
+            let keyBlob: Data = writer.data(secret: secret)
+            let curveData: Data = writer.curveType(for: secret.algorithm, length: secret.keySize).data(using: .utf8)!
```
Comment (maxgoedjen) [inline Sources/Packages/Sources/SecretAgentKit/Agent.swift:94]:
```suggestion
            let keyBlob = writer.data(secret: secret)
            let curveData = writer.curveType(for: secret.algorithm, length: secret.keySize).data(using: .utf8)!
```

**Review by maxgoedjen — APPROVED** [review]

**Pull request merged**